driver/ppm_fillers.c

```diff
@@ -3298,20 +3298,25 @@ static int poll_parse_fds(struct event_filler_arguments *args, bool enter_event)
 	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
 
 	fds = (struct pollfd *)args->str_storage;
+
+	/* We don't want to discard the whole event if the pointer is null.
+	 * Setting `nfds = 0` we will just push to userspace the number of fds read,
+	 * in this case `0`.
+	 */
 #ifdef CONFIG_COMPAT
 	if (!args->compat) {
 #endif
 		if (unlikely(ppm_copy_from_user(fds, (const void __user *)val, nfds * sizeof(struct pollfd))))
-			return PPM_FAILURE_INVALID_USER_MEMORY;
+			nfds = 0;
 #ifdef CONFIG_COMPAT
 	} else {
 		if (unlikely(ppm_copy_from_user(fds, (const void __user *)compat_ptr(val), nfds * sizeof(struct pollfd))))
-			return PPM_FAILURE_INVALID_USER_MEMORY;
+			nfds = 0;
 	}
 #endif
 
 	pos = 2;
-	targetbuf = args->str_storage + nfds * sizeof(struct pollfd) + pos;
+	targetbuf = args->str_storage + nfds * sizeof(struct pollfd);
```

Comment:
Please take a deeper look here! I'm quite sure this was a bug but never say never!

Reply:
Wow it seems a bad bug to me too :/